pkgs/by-name/ro/roomarranger/package.nix

```
@@ -0,0 +1,90 @@
+{
+  autoPatchelfHook,
+  fetchurl,
+  gtk3,
+  lib,
+  makeDesktopItem,
+  qt6,
+  stdenv,
+}:
+
+let
+  desktopItem = makeDesktopItem rec {
+    type = "Application";
+    exec = "roomarranger";
+    name = "roomarranger";
+    desktopName = "Room Arranger";
+    genericName = "Design your room, office, apartment, house.";
+    icon = "roomarranger-icon";
+    terminal = false;
+    categories = [ "Graphics" ];
+    mimeTypes = [
+      "application/com.roomarranger.project"
+      "application/com.roomarranger.object"
+    ];
+  };
+in
+
+stdenv.mkDerivation rec {
+  pname = "roomarranger";
+  version = "10.0.1";
+
+  src = fetchurl {
+    url = "https://f000.backblazeb2.com/file/rooarr/rooarr1001-linux64.tar.gz";
+    hash = "sha256-OwJSOfyTQinVKzrJftpFa5NN1kGweBezedpL2aE4LbE=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    gtk3
+    desktopItem
+    qt6.qt3d
+    qt6.qtwayland
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib $out/bin
+    cp -r rooarr-bin/* $out/lib/
+
+    # Remove bundled dynamic libraries
+    rm -f $out/lib/*.so*
```

I think there's some libraries in subdirectories, too, so we might want a slightly more expansive glob pattern:

```suggestion
    rm -f $out/lib/**/*.so*
```

---

when i do that, the app doesn't start anymore.
```
qt.qpa.plugin: Could not load the Qt platform plugin "wayland" in "" even though it was found.
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: wayland, wayland-egl, minimal, minimalegl, vkkhrdisplay, vnc, offscreen, eglfs, linuxfb, xcb.

[1]    41577 IOT instruction (core dumped)  ./result/bin/roomarranger
```

which is weird to me, because the build output clearly states that it finds all these dependencies, e.g. here:
```
    libxcb.so.1 -> found: /nix/store/nnq0lx0z6ncwsdmsn1prgh1d1m3hdp5i-libxcb-1.17.0/lib
```

any idea?

---

BTW, this also happens if i don't delete any of the `.so` files at all. shouldn't that also be the normal behavior? i thought `autoPatchelfHook` is exactly there so it fixes these precompiled binaries.

---

Hm, that's odd. It says it can _find_ the plugins but not load them:

> Could not load the Qt platform plugin "wayland" in "" even though it was found.

I'm not super familiar with Qt apps but maybe you need another environment variable?

Also, I remember now that there's a couple `.so` files that I couldn't find substitutes for in Nixpkgs, like `libcastleengine.so` IIRC. Maybe you need to delete those?

It's probably fine to just keep the `.so` files in the distribution at least for now, if it's too annoying to figure out how to get it working without them.

---

hey @9999_years, i've tried around a bit and it works if i only remove the dynamic libs that refer to Qt major version 6. if i leave all the libs in, it also won't start.

it seems to be an issue that this version of roomarranger was built with Qt 6.8.1, whereas the version on nixpkgs is already 6.8.2. as far as i've learned now, Qt is especially picky with its versions. i guess you could use an overlay to address this, but i'm afraid that we'll go through dependency hell then.

so, i kind of don't want to fiddle around with this any more than necessary. if there's a Qt expert out there who knows the best way to cleanly fix this, i'm more than happy to apply it. but the way it is **now** works good enough for most people i'd say, also the desktop entry works now.

please let me know if there's anything else you'd want to see in the PR. and again, thanks for your time to do these reviews. :)